### PR TITLE
Hum 852 Support generic S3 responses

### PR DESCRIPTION
**Pull request by creiht**
Note: probably best to merge this after HUM-750

### CODE REVIEW

File: proxyserver/middleware/s3api.go [hunk 1]
```diff
@@ -87,29 +102,64 @@ func NewS3Error() *s3Error {
...
-	w.WriteHeader(statusCode)
-	w.Write(output)
+	w.writer.WriteHeader(statusCode)
+}
 
+func (w *s3ResponseWriterWrapper) Write(buf []byte) (int, error) {
+	if !w.hijack {
+		return w.writer.Write(buf)
+	} else {
+		// TODO: Do we need to check to make sure everything gets written?
```
Comment (redbo) [inline proxyserver/middleware/s3api.go]: You shouldn't need to, the io.Writer interface says writes always complete in full or return an error.

Comment (creiht) [reply proxyserver/middleware/s3api.go]: Oh yeah, I had looked that up yesterday as well, thanks.

File: proxyserver/middleware/s3api.go [hunk 2]
```diff
@@ -87,29 +102,64 @@ func NewS3Error() *s3Error {
...
-		srv.SimpleErrorResponse(w, http.StatusInternalServerError, err.Error())
-		return
+}
+
+func (w *s3ResponseWriterWrapper) Header() http.Header {
+	return w.writer.Header()
+}
+
+func (w *s3ResponseWriterWrapper) WriteHeader(statusCode int) {
+	if statusCode/200 != 1 {
```
Comment (corystone) [inline proxyserver/middleware/s3api.go]: Super nitpick. I think we usually write this statusCode/100 != 2.

**Comment by corystone** [comment]
"You need to run 'make fmt'"

**Pull request merged**